connector/server/survivalanalysis/survival_query.go

```diff
@@ -0,0 +1,413 @@
+package survivalserver
+
+import (
+	"encoding/base64"
+	"fmt"
+	"strings"
+	"sync"
+	"time"
+
+	"github.com/pkg/errors"
+	"github.com/sirupsen/logrus"
+
+	querytoolsserver "github.com/ldsec/medco/connector/server/querytools"
+	utilcommon "github.com/ldsec/medco/connector/util/common"
+	utilserver "github.com/ldsec/medco/connector/util/server"
+	"github.com/ldsec/medco/connector/wrappers/i2b2"
+	"github.com/ldsec/medco/connector/wrappers/unlynx"
+
+	"github.com/ldsec/medco/connector/restapi/models"
+	"github.com/ldsec/medco/connector/restapi/server/operations/survival_analysis"
+)
+
+// Query holds the ID of the survival analysis, its parameters and a pointer to its results
+type Query struct {
+	UserID              string
+	UserPublicKey       string
+	QueryName           string
+	CohortName          string
+	SubGroupDefinitions []*survival_analysis.SurvivalAnalysisParamsBodySubGroupDefinitionsItems0
+	TimeLimit           int
+	TimeGranularity     string
+	StartConcept        string
+	StartModifier       string
+	EndConcept          string
+	EndModifier         string
+	Result              *struct {
+		Timers    utilcommon.Timers
+		EncEvents EventGroups
+	}
+}
+
+// NewQuery query constructor
+func NewQuery(UserID,
+	QueryName,
+	UserPublicKey string,
+	CohortName string,
+	SubGroupDefinitions []*survival_analysis.SurvivalAnalysisParamsBodySubGroupDefinitionsItems0,
+	TimeLimit int,
+	TimeGranularity string,
+	StartConcept string,
+	StartModifier string,
+	EndConcept string,
+	EndModifier string) *Query {
+	res := &Query{
+		UserPublicKey:       UserPublicKey,
+		UserID:              UserID,
+		QueryName:           QueryName,
+		CohortName:          CohortName,
+		SubGroupDefinitions: SubGroupDefinitions,
+		TimeLimit:           TimeLimit,
+		TimeGranularity:     TimeGranularity,
+		StartConcept:        StartConcept,
+		StartModifier:       StartModifier,
+		EndConcept:          EndConcept,
+		EndModifier:         EndModifier,
+		Result: &struct {
+			Timers    utilcommon.Timers
+			EncEvents EventGroups
+		}{}}
+	res.Result.Timers = make(map[string]time.Duration)
+
+	return res
+}
+
+// Execute runs the survival analysis query
+func (q *Query) Execute() error {
```

Comment:
This high-level function needs some `info` log calls to make aware of what is happening during the execution of the query.
Also it would make sense to have it broken down into smaller chunks.